Software/MainLoop.cs

```diff
@@ -66,14 +74,16 @@ public static void Run(CancellationToken exitToken)
                 int transferredIn;
                 byte[] readBuffer = new byte[38];
                 ErrorCode ecRead = reader.Transfer(readBuffer, 0, readBuffer.Length, 100, out transferredIn);
-                if (ecRead != ErrorCode.None) throw new Exception("Submit Async Read Failed.");
+                if (ecRead != ErrorCode.None)
+                {
+                    _logger.Error($"Submit Async Read Failed. ErrorCode: {ecRead}");
+                    return;
```

Comment:
this should probably do that cancellation token source thingee

Reply:
Good point.